From 9072ef03f8f43cd16ec1a2e639391044d335e8c7 Mon Sep 17 00:00:00 2001
From: Alexandru Costache <alexandru@balena.io>
Date: Fri, 18 Oct 2019 14:54:11 +0200
Subject: [PATCH] fbcon: ignore events for rpi sense framebuffer

Various crashes occurr in the fb core
when RPI Sensehat's 8x8 pixel display fb is
bound to a fbconsole.

Don't attach consoles to this particular
framebuffer to avoid kernel panics in the
fb core.

Crashes won't occur if another framebuffer
was previously created, because console does
not get switched to the sensehat in that case.

Upstream-status: Inappropriate [configuration]
Signed-off-by: Alexandru Costache <alexandru@balena.io>
---
 drivers/video/fbdev/core/fbcon.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/video/fbdev/core/fbcon.c b/drivers/video/fbdev/core/fbcon.c
index f7ac64d692b2..ff480d4d7144 100644
--- a/drivers/video/fbdev/core/fbcon.c
+++ b/drivers/video/fbdev/core/fbcon.c
@@ -3338,6 +3338,15 @@ static int fbcon_event_notify(struct notifier_block *self,
 				  action == FB_EVENT_FB_UNREGISTERED))
 		goto done;
 
+	/* If no other framebuffer is availabe, then sensehat's
+	 * will be used. But it's an 8x8 pixels display, not meant for this.
+	 * Ignore all events for rpi-sense fb and don't attach a console to it.
+	 */
+	if (info && !strncmp(info->fix.id, "RPi-Sense FB", 12))
+	{
+		goto done;
+	}
+
 	switch(action) {
 	case FB_EVENT_SUSPEND:
 		fbcon_suspended(info);
-- 
2.17.1

